packages/cheqd-sdk/src/dids/CheqdDidRegistrar.ts

```diff
@@ -0,0 +1,423 @@
+import type {
+  AgentContext,
+  DidRegistrar,
+  DidCreateOptions,
+  DidCreateResult,
+  DidDeactivateResult,
+  DidUpdateResult,
+  DidDocument,
+  Jwk,
+  VerificationMethod,
+} from '@aries-framework/core'
+import type { CheqdNetwork, DIDDocument, DidStdFee, TVerificationKey, VerificationMethods } from '@cheqd/sdk'
+import type { SignInfo } from '@cheqd/ts-proto/cheqd/did/v2'
+
+import {
+  DidDocumentRole,
+  DidRecord,
+  DidRepository,
+  KeyType,
+  Key,
+  Buffer,
+  isValidPrivateKey,
+} from '@aries-framework/core'
+import { MethodSpecificIdAlgo, createDidVerificationMethod } from '@cheqd/sdk'
+import { MsgCreateResourcePayload } from '@cheqd/ts-proto/cheqd/resource/v2'
+import { fromString, toString } from 'uint8arrays'
+import { v4 } from 'uuid'
```

Comment:
You can import `utils` from '@aries-framework/core` which exposes an `uuid` method. This will generate an v4 uuid